src/Framework/AspNetCoreAnalyzers/test/Authorization/AddAuthorizationBuilderTests.cs

```diff
@@ -0,0 +1,585 @@
+    .AddPolicy(""AtLeast21"", policy =>
+        policy.Requirements.Add(new MinimumAgeRequirement(21)));
+";
+
+        await VerifyCodeFix(source, new[] { diagnostic }, fixedSource);
+    }
+
+    [Fact]
+    public async Task AddAuthorization_IsTheLastCallInChain_FixedWithAddAuthorizationBuilder()
+    {
+        var diagnostic = new DiagnosticResult(DiagnosticDescriptors.UseAddAuthorizationBuilder)
+           .WithLocation(0)
+           .WithMessage(Resources.Analyzer_UseAddAuthorizationBuilder_Message);
+
+        var source = @"
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.Extensions.DependencyInjection;
+
+var builder = WebApplication.CreateBuilder(args);
+
+{|#0:builder.Services.AddRouting()
+    .AddAuthorization(options =>
+    {
+        options.AddPolicy(""AtLeast21"", policy =>
+            policy.Requirements.Add(new MinimumAgeRequirement(21)));
+    })|};
+";
+
+        var fixedSource = @"
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.Extensions.DependencyInjection;
+
+var builder = WebApplication.CreateBuilder(args);
+
+builder.Services.AddRouting()
+    .AddAuthorizationBuilder()
+    .AddPolicy(""AtLeast21"", policy =>
+            policy.Requirements.Add(new MinimumAgeRequirement(21)));
+";
+
+        await VerifyCodeFix(source, new[] { diagnostic }, fixedSource);
+    }
+
+    [Fact]
+    public async Task AddAuthorization_IsNotTheLastCallInChain_NoDiagnostic()
+    {
+        var source = @"
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.Extensions.DependencyInjection;
+
+var builder = WebApplication.CreateBuilder(args);
+
+builder.Services.AddAuthorization(options =>
+{
+    options.AddPolicy(""AtLeast21"", policy =>
+        policy.Requirements.Add(new MinimumAgeRequirement(21)));
+})
+.AddAuthentication();
+";
+
+        await VerifyNoCodeFix(source);
+    }
+
+    [Fact]
+    public async Task AuthorizationOptions_DefaultPolicyAccess_NoDiagnostic()
+    {
+        var source = @"
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.Extensions.DependencyInjection;
+
+var builder = WebApplication.CreateBuilder(args);
+
+builder.Services.AddAuthorization(options =>
+{
+    _ = options.DefaultPolicy;
+    options.AddPolicy(""AtLeast21"", policy =>
+    {
+        policy.Requirements.Add(new MinimumAgeRequirement(21));
+    });
+});
+";
+
+        await VerifyNoCodeFix(source);
+    }
+
+    [Fact]
+    public async Task AuthorizationOptions_FallbackPolicyAccess_NoDiagnostic()
+    {
+        var source = @"
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.Extensions.DependencyInjection;
+
+var builder = WebApplication.CreateBuilder(args);
+
+builder.Services.AddAuthorization(options =>
+{
+    _ = options.FallbackPolicy;
+    options.AddPolicy(""AtLeast21"", policy =>
```

Comment:
I do like that this doesn't try to use the AuthorizationBuilder unless the configure AuthorizationOptions callback can be completely removed 👍

Reply:
Thank you! I wanted to make the analyzer/code fix as "cautious" as possible, so it won't inadvertently break code that uses any `AuthorizationOptions`-specific APIs.